vulture_os/authentication/migrations/0012_auto_20210615_1732.py

```diff
@@ -0,0 +1,90 @@
+# Generated by Django 3.0.5 on 2021-06-15 17:32
+
+import authentication.user_scope.models
+import bson.objectid
+from django.db import migrations, models
+import django.db.models.deletion
+import djongo.models.fields
+
+
+user_scopes = {}
+
+def save_user_scope(apps, schema_editor):
+    portal_model = apps.get_model("authentication", "UserAuthentication")
+    db_alias = schema_editor.connection.alias
+    portal_objects = portal_model.objects.using(db_alias)
+
+    for portal in portal_objects.all():
+        user_scopes[portal.name] = portal.repo_attributes
+
+
+def restore_user_scope(apps, schema_editor):
+    scope_model = apps.get_model("authentication", "UserScope")
+    portal_model = apps.get_model("authentication", "UserAuthentication")
+    db_alias = schema_editor.connection.alias
+    scope_objects = scope_model.objects.using(db_alias)
+    portal_objects = portal_model.objects.using(db_alias)
+
+    for portal_name, scope in user_scopes.items():
+        user_scope = scope_objects.create(name=f"Scope {portal_name}")
```

Comment:
```suggestion
        user_scope = scope_objects.create(name=f"Scope_{portal_name}")
```

Reply:
I just removed the form cleaning instead of modifying the migration.
No need to replace spaces by underscores in those objects, there is no associated haproxy templates.